eng/dotnet-build/build.sh

```diff
@@ -89,6 +89,7 @@ fi
 
 properties="$properties /p:Configuration=$configuration"
 properties="$properties /p:DotNetBuildRepo=true"
+properties="$properties /p:DotNetBuildFromSource=true"
```

Comment:
I don't know how this is supposed to work. @mmitche made this file. So I'll defer to him.

Reply:
This should not be required. This repo no longer uses the legacy properties. https://github.com/NuGet/NuGet.Client/pull/5795/files#diff-a15425150d4f18ff5c2922f79b5344487dc77935032712545d2cf4a5982557c6R87 just above here is passing the correct source-only switch.

Reply:
I'm confused.  This insertion failed: https://github.com/dotnet/sdk/pull/40791

Because its building our repo and hits
```
/vmr/src/nuget-client/Directory.Build.targets(3,3): error MSB4019: The imported project "/vmr/src/nuget-client/eng/source-build/ExcludeFromSourceBuild.AfterCommonTargets.targets" was not found.
```

So the fix was to remove the bad import
```xml
<Import Project="eng\source-build\ExcludeFromSourceBuild.AfterCommonTargets.targets" 
Condition="'$(DotNetBuildFromSource)' == 'true'" />
```

Where the implication was that source-build was busted.  But it sounds like NuGet actually broke VMR?  I can revert the changes to these scripts and still fix the original issue.  The changes I made was to help us prevent making bad changes.

Reply:
Some repositories in the VMR are still on 8.0 versions of arcade, where DotNetBuildFromSource is the only property available. So the VMR passes both DotNetBuildSourceOnly and DotNetBuildFromSource to all repos to ensure compatibility. It's used in some repos, but not in others.

This is why you see a failure there. But because NuGet isn't on arcade, I was able to upgrade to the 9.0 control set when I enabled VMR builds a month or so ago. However, the property is still getting passed by the VMR, which is why you saw the failure. Its only use was in that erroneously added line.

Reply:
Okay, thanks for the clarity.  I'll revert the changes to these scripts.